cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_cloud_provider.go

```diff
@@ -184,3 +211,23 @@ func BuildHuaweiCloud(opts config.AutoscalingOptions, do cloudprovider.NodeGroup
 
 	return newCloudProvider(opts, do, rl)
 }
+
+func buildAsgFromSpec(specStr string, asgs []AutoScalingGroup, manager CloudServiceManager) (*AutoScalingGroup, error) {
+	spec, err := dynamic.SpecFromString(specStr, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse node group spec: %v", err)
+	}
+
+	for _, asg := range asgs {
+		if asg.groupName == spec.Name {
+			return &AutoScalingGroup{
+				cloudServiceManager: manager,
+				groupName:           asg.groupName,
+				groupID:             asg.groupID,
+				minInstanceNumber:   asg.minInstanceNumber,
+				maxInstanceNumber:   asg.maxInstanceNumber,
+			}, nil
+		}
+	}
+	return nil, fmt.Errorf("No auto scaling group found, spec: %s", spec.Name)
```

Comment:
Error string should not be capitalized or end with punctuation.

More details please refer to [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).